### PR TITLE
Function to get attribute using path

### PR DESCRIPTION
**Pull request by tisnik**
For the new tests we need to check if the object returned from API (via JSON) contains a given attribute/value stored in nested structure of objects and lists. This helper function allow us to develop these tests quite easily.

An example (component analysis result), the path for the given attribute might look like:

"data/0/package/ecosystem/0"

### CODE REVIEW

**Comment by tisnik** [comment]
[test this]

**Comment by tisnik** [comment]
[test this]

File: integration-tests/features/steps/common.py [hunk 1]
```diff
@@ -479,3 +479,31 @@ def check_stack_analyses_response(context, url):
...
+    get_value_using_path({"x" : {"y" : "z"}}, "x/y")) -> "z"
+    get_value_using_path(["x", "y", "z"], "0"))       -> "x"
+    get_value_using_path(["x", "y", "z"], "1"))       -> "y"
+    get_value_using_path({"key1" : ["x", "y", "z"],
+                          "key2" : ["a", "b", "c", "d"]}, "key1/1")) -> "y"
+    get_value_using_path({"key1" : ["x", "y", "z"],
+                          "key2" : ["a", "b", "c", "d"]}, "key2/1")) -> "b"
+    """
+
+    keys = path.split("/")
```
Comment (fridex) [inline integration-tests/features/steps/common.py]: What about passing directly an arrays/tuple? A key can have "/" in that case.

File: integration-tests/features/steps/common.py [hunk 2]
```diff
@@ -479,3 +479,31 @@ def check_stack_analyses_response(context, url):
...
+    for key in keys:
+        if key.isdigit():
+            object = object[int(key)]
```
Comment (fridex) [inline integration-tests/features/steps/common.py]: Shadowed Python's `object`.

File: integration-tests/features/steps/common.py [hunk 3]
```diff
@@ -479,3 +479,31 @@ def check_stack_analyses_response(context, url):
...
+        else:
+            object = object[key]
+    return object
```
Comment (fridex) [inline integration-tests/features/steps/common.py]: Missing default return value. If the for-loop is not done, `object` is not defined.

File: integration-tests/features/steps/common.py (hunk 1 above)
Comment (tisnik) [inline integration-tests/features/steps/common.py]: Well I was thinking about it, but how to specify the tuple in the test definition? In brackets and by using commas as separator?

File: integration-tests/features/steps/common.py (hunk 2 above)
Comment (tisnik) [inline integration-tests/features/steps/common.py]: oh you are right, thanks

File: integration-tests/features/steps/common.py (hunk 3 above)
Comment (tisnik) [inline integration-tests/features/steps/common.py]: would not it return the original object?

Comment (fridex) [inline integration-tests/features/steps/common.py]: Ah, you are right, its parameter. +1

File: integration-tests/features/steps/common.py (hunk 1 above)
Comment (fridex) [inline integration-tests/features/steps/common.py]: Ah, OK - that was why I was curious about the actual usage. Now it makes sense.

File: integration-tests/features/steps/common.py (hunk 2 above)
Comment (tisnik) [inline integration-tests/features/steps/common.py]: fixed this one

**Pull request merged**